src/ReflectionExtension.php

```diff
@@ -0,0 +1,76 @@
+<?php
+/**
+ * Parser Reflection API
+ *
+ * @copyright Copyright 2015, Lisachenko Alexander <lisachenko.it@gmail.com>
+ *
+ * This source file is subject to the license that is bundled
+ * with this source code in the file LICENSE.
+ */
+
+namespace Go\ParserReflection;
+
+use ReflectionExtension as BaseReflectionExtension;
+
+/**
+ * Returns AST-based reflections from extensions.
+ */
+class ReflectionExtension extends BaseReflectionExtension implements IReflection
+{
+    /**
+     * @var null|ReflectionClass[] ParsedReflection wrapped classes.
+     */
+    private $classes;
+
+    /**
+     * @var null|ReflectionFunction[] ParsedReflection wrapped functions.
+     */
+    private $functions;
+
+    /**
+     * Has extension been loaded by PHP.
+     *
+     * @return true
+     *     Enabled extensions are always loaded.
+     */
+    public function wasIncluded()
+    {
+        return true;
+    }
+
+    /**
+     * Returns list of reflection classes
+     *
+     * @return array|\ReflectionClass[]
+     */
+    public function getClasses()
+    {
+        if (!isset($this->classes)) {
+            $classRefs = parent::getClasses();
+            $this->classes = [];
+            foreach ($classRefs as $origKey => $eachClass) {
```

Comment:
Please rename:

* `$origKey` into `$index`
* `$eachClass` into `$classReflection`

---

I think rename makes names easier to understand.